### ISSUE TITLE
Custom loss usage by name: not working in V3 but worked in V2

### ISSUE THREAD

**Issue by shkarupa-alex** [keras-team/keras#20112]
Default use case for custom loss in V2 (for me) was:

1. Register custom loss with register_keras_serializable
2. Use it in model compile by name compile(loss="MyPackage>MyLoss")

This worked in Keras V2, but not working in Keras V3. Here is a simple test https://colab.research.google.com/drive/1R3z3fFhV7NN1Oa_HJYp60LsneoxjOlvj?usp=sharing

**Comment by chunduriv** [comment]
@shkarupa-alex,

Unfortunately, I don't have access to Colab. Could you please grant us access?

**Comment by shkarupa-alex** [comment]
shared

**Comment by nkovela1** [comment]
Hi @shkarupa-alex, thanks for the Colab! Could you try `keras.saving.get_registered_object(name)` with the name "MyPackage>MyLoss"?

**Comment by shkarupa-alex** [comment]
> Hi @shkarupa-alex, thanks for the Colab! Could you try `keras.saving.get_registered_object(name)` with the name "MyPackage>MyLoss"?

Added to the same colab notebook: `keras.saving.get_registered_object("MyPackage>MyLoss")` returns custom loss class

**Comment by maitry63** [comment]
HI @shkarupa-alex , I reproduce the issue in  latetst keras  version and issue doesn´t reproduce anymore. 
Attached [gist](https://colab.sandbox.google.com/gist/maitry63/89c56e5a8ae0c5dd0bf0bc57ca2fd53d/20112.ipynb) for the reference.
